src/solvers/sat/satcheck_minisat2.cpp

```diff
@@ -371,3 +371,11 @@ bool satcheck_minisat_simplifiert::is_eliminated(literalt a) const
 
   return solver->isEliminated(a.var_no());
 }
+
+bool satcheck_minisat_simplifiert::set_parameter(const char *const param)
+{
+  int p_len = strlen(param) + 2;
+  char *p = new char[p_len];
```

Comment:
Is it intentional that this throws away the `char[]` allocation without deletion? If not, suggest using `std::string` or similar to hold and manage the lifespan of the memory.

Reply:
No, not intentional, it's a mistake. Actually I should be using std::string throughout the changes. Will fix. Thanks for the feedback @smowton !